lib/internal/child_process.js

```diff
@@ -699,6 +701,23 @@ function setupChannel(target, channel, serializationMode) {
     });
   });
 
+  target.on('newListener', function() {
+
+    process.nextTick(() => {
+      if (!target.channel || !target.listenerCount('message'))
+        return;
+
+      const messages = target.channel[kPendingMessages];
+      if (!messages.length) return;
+
+      for (const messageParams of messages) {
+        ReflectApply(target.emit, target, messageParams);
```

Comment:
`for of` loops rely on globally mutable `Array.prototype[Symbol.iterator]`, prefer the classic `for(;;)` loop:
```suggestion
      const { length } = messages;
      if (length === 0) return;

      for (let i = 0; i < length; i++) {
        ReflectApply(target.emit, target, messages[i]);
```

Reply:
fixed